src/client/signatures/rc_crypto_verifier.rs

```diff
@@ -12,13 +12,29 @@ pub struct RcCryptoVerifier {}
 impl RcCryptoVerifier {}
 
 impl Verification for RcCryptoVerifier {
-    fn verify(&self, collection: &Collection) -> Result<(), SignatureError> {
+    fn verify(&self, collection: &Collection, root_hash: &str) -> Result<(), SignatureError> {
         debug!("Verifying using x509-parser and rc_crypto");
 
         // Get public key from certificate (PEM from `x5u` field).
         let pem_bytes = self.fetch_certificate_chain(&collection)?;
 
-        let public_key_bytes = x509::extract_public_key(pem_bytes)?;
+        let pems = x509::parse_certificate_chain(&pem_bytes)?;
+
+        let certs: Vec<x509::X509Certificate> = pems
+            .iter()
+            .map(|pem| match x509::parse_x509_certificate(&pem) {
+                Ok(cert) => Ok(cert),
+                Err(e) => Err(e),
+            })
+            .collect::<Result<Vec<x509::X509Certificate>, _>>()?;
```

Comment:
I guess we need to copy (or re-use) the `// Verify signature chain` logic from the `ring` implementation into this one as well.

Without thinking too deeply about it, would it make sense to factor that out into a shared default helper method on the Trait? Along the lines of:

```
pub trait Verification {
    // ... existing default methods here ...

    fn verify_certificate_chain(&self, certs: Vec<x509::X509Certificate>, root_hash: &str) -> Result<x509::X509Certificate, SignatureError> {
        // Do all the parsing and stuff.
        // Call methods on `self` to delegate to verification of each signature to the trait implemetor
        // Return the leaf certificate by use by calling code.
    }
```

It seems like *most* of the code there is parsing stuff that could usefully be shared between implementations.